File: Diappetes_App/app/src/main/java/com/serkanonder/diappetes/BluetoothMainActivity.java
package com.serkanonder.diappetes;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.bluetooth.BluetoothAdapter;
import android.bluetooth.BluetoothDevice;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.ListView;
import android.widget.TextView;
import android.widget.Toast;
import java.util.ArrayList;
import java.util.Set;

public class BluetoothMainActivity extends AppCompatActivity {

    BluetoothAdapter mBluetoothAdapter;
    Button btnConnection;
    Button btnPairDevices;
    ListView lwPairedList;
    private Set<BluetoothDevice> pairedDevices;
    public static String EXTRA_ADDRESS = "device_address";


    @SuppressLint("MissingInflatedId")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mBluetoothAdapter = BluetoothAdapter.getDefaultAdapter();
        btnConnection = findViewById(R.id.btnConnection);
        btnPairDevices = findViewById(R.id.btnPairDevices);
        lwPairedList = findViewById(R.id.lwPairedList);

        btnConnection.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                toogleBluetooth();
            }
        });

        btnPairDevices.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                listDevices();
            }
        });
    }

    @SuppressLint("MissingPermission")
    private void listDevices() {
        pairedDevices = mBluetoothAdapter.getBondedDevices();
        ArrayList list = new ArrayList();

        if(pairedDevices.size() > 0){
            for(BluetoothDevice bt: pairedDevices){
                list.add(bt.getName() + "\n" + bt.getAddress());
            }
        } else {
            Toast.makeText(getApplicationContext(), "Eşleşmiş Cihaz Yok", Toast.LENGTH_SHORT).show();
        }

        final ArrayAdapter adapter = new ArrayAdapter(this, android.R.layout.simple_list_item_1, list);
        lwPairedList.setAdapter(adapter);
        lwPairedList.setOnItemClickListener(selectDevice);

    }

    public AdapterView.OnItemClickListener selectDevice = new AdapterView.OnItemClickListener() {
        @Override
        public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
            String info = ((TextView) view).getText().toString();
            String address = info.substring(info.length()-17);

            Intent comIntent = new Intent(BluetoothMainActivity.this,CommunicationActivity.class);
            comIntent.putExtra(EXTRA_ADDRESS, address);
            startActivity(comIntent);
        }
    };

    @SuppressLint("MissingPermission")
    private void toogleBluetooth() {
        if(mBluetoothAdapter == null){
            Toast.makeText(getApplicationContext(), "Bluetooth Cihazı Bulunamadı", Toast.LENGTH_SHORT).show();
        }

        if(!mBluetoothAdapter.isEnabled()){
            Intent enableBTIntent = new Intent(BluetoothAdapter.ACTION_REQUEST_ENABLE);
            startActivity(enableBTIntent);
            btnConnection.setText("Bluetoothu Kapat");
        }

        if(mBluetoothAdapter.isEnabled()){
            mBluetoothAdapter.disable();
            btnConnection.setText("Bluetoothu Aç");
        }



    }
}

File: Diappetes/app/src/main/java/com/serkanonder/diappetes/BluetoothMainActivity.kt
package com.serkanonder.diappetes

import android.annotation.SuppressLint
import android.app.Activity
import android.bluetooth.BluetoothAdapter
import android.bluetooth.BluetoothDevice
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.*
import android.widget.AdapterView.OnItemClickListener
import androidx.appcompat.app.AppCompatActivity
import com.serkanonder.diappetes.databinding.ActivityBluetoothMainBinding
import kotlinx.android.synthetic.main.activity_bluetooth_main.*

class BluetoothMainActivity : AppCompatActivity() {
    lateinit var binding: ActivityBluetoothMainBinding

    private var bluetoothAdapter: BluetoothAdapter? = null
    private lateinit var pairedDevices: Set<BluetoothDevice>
    private val REQUEST_ENABLE_BLUETOOTH = 1
    companion object {
        var EXTRA_ADDRESS: String = "device_address"
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityBluetoothMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        bluetoothAdapter = BluetoothAdapter.getDefaultAdapter()
        if (bluetoothAdapter == null){
            Toast.makeText(this,"This device doesn't support Bluetooth", Toast.LENGTH_SHORT).show()
            return
        }
        binding.btnPairDevices.setOnClickListener{pairedDeviceList()}
    }
    @SuppressLint("MissingPermission")
    private fun pairedDeviceList(){
        pairedDevices = bluetoothAdapter!!.bondedDevices
        val list : ArrayList<BluetoothDevice> = ArrayList()
        if(!pairedDevices.isEmpty()){
            for(device: BluetoothDevice in pairedDevices){
                list.add(device)
                Toast.makeText(this,"Showing Paired Devices", Toast.LENGTH_SHORT).show()
            }
        }else{
            Toast.makeText(this,"No Paired Bluetooth Devices Found.", Toast.LENGTH_SHORT).show()
        }
        val adapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, list)
        lwPairedList.adapter = adapter
        lwPairedList.onItemClickListener = AdapterView.OnItemClickListener(){_, _, position, _ ->
            val device: BluetoothDevice = list[position]
            val address: String = device.address
            val intent = Intent(this, RecevingDataActivity::class.java)
            intent.putExtra(EXTRA_ADDRESS, address)
            startActivity(intent)
        }

    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == REQUEST_ENABLE_BLUETOOTH) {
            if (resultCode == Activity.RESULT_OK) {
                if (bluetoothAdapter!!.isEnabled) {
                    Toast.makeText(this, "Bluetooth has been enabled", Toast.LENGTH_SHORT).show()
                } else {
                    Toast.makeText(this, "Bluetooth has been disabled", Toast.LENGTH_SHORT).show()
                }
            } else if (resultCode == Activity.RESULT_CANCELED) {
                Toast.makeText(this, "Bluetooth enabling has been canceled", Toast.LENGTH_SHORT).show()
            }
        }
    }
}